src/hidimstat/dcrt.py

```diff
@@ -11,265 +11,379 @@
-        variables along with the list of selected variables.
-    joblib_versobe : int, default=0
-       The verbosity level of joblib: if non zero, progress messages are
-       printed. Above 50, the output is sent to stdout. The frequency of the
-       messages increases with the verbosity level. If it more than 10, all
-       iterations are reported.
+        Number of parallel jobs
+    joblib_verbose : int, default=0
+        Verbosity level
+    fit_y : bool, default=False
+        Whether to fit y using selected features
     ntree : int, default=100
-        The number of trees for the distillation using the Random Forest
-        learner.
-    problem_type : str, default='regression'
-        A classification or a regression problem.
-    random_state : int, default=2023
-        Fixing the seeds of the random generator.
+        Number of trees for random forest
+    problem_type : {'regression', 'classification'}, default='regression'
+        Type of learning problem
+    random_state : int, default=2022
+        Random seed
 
     Returns
     -------
-    selected : 1D array, int
-        The vector of index of selected variables.
-    pvals: 1D array, float
-        The vector of the corresponding p-values.
-    ts: 1D array, float
-        The vector of the corresponding test statistics.
+    selection_features : ndarray
+        Boolean mask of selected features
+    X_res : ndarray
+        Residuals after distillation
+    sigma2_X : ndarray
+        Estimated residual variances
+    y_res : ndarray
+        Response residuals
 
     References
     ----------
     .. footbibliography::
     """
     if centered:
-        X = StandardScaler().fit_transform(X)
+        X_ = StandardScaler().fit_transform(X)
+    else:
+        X_ = X
+    y_ = y  # avoid modifying the original y
 
-    _, n_features = X.shape
+    _, n_features = X_.shape
 
+    ## Screening of variables for accelarate dCRT
     if estimated_coef is None:
-        if loss == "least_square":
-            clf = LassoCV(
-                cv=cv,
-                n_jobs=n_jobs,
-                n_alphas=n_alphas * 2,
-                tol=1e-6,
-                fit_intercept=False,
-                random_state=0,
-                max_iter=max_iter,
-            )
-        else:
-            raise ValueError(f"{loss} loss is not supported.")
-        clf.fit(X, y)
+        # base on the Theorem 2 of :cite:`liu2022fast`, the rule of screening
+        # is based on a cross-validated lasso
+        clf, alpha = _fit_lasso(
+            X_,
+            y_,
+            alpha=alpha,
+            alphas=alphas,
+            n_alphas=n_alphas * 2,  # TODO: Why * 2 ?
+            cv=cv,
+            n_jobs=n_jobs,
+            random_state=random_state,
+            alpha_max_fraction=alpha_max_fraction,
+            # other argument for the cross validation not used  by other fitting
+            # TODO: What is the reason for this specific options here?
```

Comment:
Do you think that these options should also be included in the other CV?

Reply:
No idea, I would have to go deep into code to understand that, which I cannot do now.

Reply:
same here.